relays/ethereum/src/ethereum_client.rs

```diff
@@ -410,18 +412,97 @@ impl EthereumHighLevelRpc for EthereumRpcClient {
 	}
 }
 
+pub const HEADERS_BATCH: usize = 4;
+
+#[derive(Debug)]
+#[cfg_attr(test, derive(Clone))]
+pub struct Headers {
```

Comment:
Docs

I think they should mention that the Solidity contract since that's driving the design, and the reason behind that